org.opent2t.sample.thermostat.superpopular/com.nest.thermostat/js/thingTranslator.js

```diff
@@ -275,6 +278,133 @@ function validateResourceGet(resourceId) {
+        var max = getMaxTemperature(providerSchema, unit);
+        if (value >= min && value <= max) {
+            return unit;
+        }
+        throw new OpenT2TError(440, "Invalid temperature (" + value + ") for unit (" + unit + " [" + min + ", " + max + "])");
+    } else {
+        var min_f = getMinTemperature(providerSchema, 'f');
+        var max_f = getMaxTemperature(providerSchema, 'f');
+        if (value >= min_f && value <= max_f) {
+            return 'f';
+        }
+        var min_c = getMinTemperature(providerSchema, 'c');
+        var max_c = getMaxTemperature(providerSchema, 'c');
+        if (value >= min_c && value <= max_c) {
+            return 'c';
+        }
+        throw new OpenT2TError(440, "Temperature outside supported range (" + value + ")");
+    }
+
+}
+
+/**
+ * Given a target temperature, compute a target 
+ * low and target high centerd around it.
+ * @param {*} resourceSchema 
+ * @param {*} providerSchema 
+ */
+function getTargetTemperatureRange(resourceSchema, providerSchema) {
+    var result = {};
+
+    var value = resourceSchema.temperature;
+    var unit = getValidatedUnit(resourceSchema, providerSchema);
+    var min = getMinTemperature(providerSchema, unit);
+    var max = getMaxTemperature(providerSchema, unit);
+
+    var targetLow = getTargetTemperatureLow(providerSchema, unit);
+    var targetHigh = getTargetTemperatureHigh(providerSchema, unit);
+
+    var range = targetHigh - targetLow;    
+    // Minimum range set by Nest API
+    if(range < 3) {
+        range = 3; 
+    }
+
+    var halfRange = Math.round(range / 2);
+    var newTargetLow = value - halfRange;
+    var newTargetHigh = value + halfRange;
+
+    // If either value is outside min/max range, adjust accordingly
+    if (newTargetLow < min) {
+        newTargetLow = min;
+        newTargetHigh = min + range;
+    }
+
+    if (newTargetHigh > max) {
+        newTargetHigh = max;
+        newTargetLow = Math.max(min, max - range);
+    }
+
+    result['target_temperature_high_' + unit] = newTargetHigh;
+    result['target_temperature_low_' + unit] = newTargetLow;
+
+    return result;
+}
+
+function getTargetTemperatureHigh(providerSchema, unit) {
+    if (unit === 'f') {
+        return providerSchema.target_temperature_high_f;
+    } else if (unit === 'c') {
+        return providerSchema.target_temperature_high_c;
+    }
+    throw new OpenT2TError(440, "Invalid temperature unit (" + unit + ")");
+}
+
+function getTargetTemperatureLow(providerSchema, unit) {
+    if (unit === 'f') {
+        return providerSchema.target_temperature_low_f;
+    } else if (unit === 'c') {
+        return providerSchema.target_temperature_low_c;
+    }
+    throw new OpenT2TError(440, "Invalid temperature unit (" + unit + ")");
+}
+
+function getMinTemperature(providerSchema, unit) {
+    if (unit === 'f') {
+        return providerSchema.is_locked ? providerSchema.locked_temp_min_f : 50;
```

Comment:
Is providerSchema.locked_temp_min_f guaranteed to be a valid numeric?
Same question for the similar statements below.

Reply:
If is_locked is set, adding a safe getInt method just in case.